File: src/main/java/tk/hiddenname/smarthome/service/hardware/impl/digital/output/DigitalDeviceService.java
package tk.hiddenname.smarthome.service.hardware.impl.digital.output;

import tk.hiddenname.smarthome.model.signal.DigitalState;
import tk.hiddenname.smarthome.service.hardware.impl.GPIOService;

public interface DigitalDeviceService extends GPIOService {

    DigitalState getState(Long id, Boolean reverse);

    DigitalState setState(Long id, Boolean reverse, Boolean newState);
}

File: src/main/kotlin/tk/hiddenname/smarthome/service/hardware/impl/digital/output/DigitalDeviceService.kt
package tk.hiddenname.smarthome.service.hardware.impl.digital.output

import tk.hiddenname.smarthome.model.signal.DigitalState
import tk.hiddenname.smarthome.service.hardware.impl.GPIOService

interface DigitalDeviceService : GPIOService {

    fun getState(id: Long, reverse: Boolean): DigitalState

    fun setState(id: Long, reverse: Boolean, newState: Boolean): DigitalState
}